### ISSUE TITLE
[web] expo-payments-stripe support

### ISSUE THREAD

**Issue by EvanBacon** [expo/expo#4046]
* [Spec](https://www.w3.org/TR/payment-request/)

**Comment by nandorojo** [comment]
Would you suggest just using stripe's web UI for now?

**Comment by RWOverdijk** [comment]
This.

**Comment by baptisteArnaud** [comment]
this

**Comment by alexanderqchen** [comment]
please

**Comment by nandorojo** [comment]
I’m using stripe’s React lib on web and it works well. I’d suggest doing that until expo adds web support. It was pretty simple. For an example of it working, see https://app.doorman.cool -> sign up -> create project -> billing screen.

**Comment by RWOverdijk** [comment]
I use elements in a webview now, too. But it's _far_ from ideal. Imo expo payments is in a useless state, also because of the use of old apis.

But I'm new to this so maybe my opinion is based on the wrong thoughts.

**Comment by RWOverdijk** [comment]
Okay, that's nice. But who uses expo and doesn't also run on mobile? The whole point of this is to have support across platforms, isn't it?

**Comment by edi** [comment]
@RWOverdijk The WEB API is not a generally available version (currently in BETA). Thefore you should not expect something meant for an API in Beta, to be fully usable or have full support.

Be patient, there are other more important APIs to implement.

**Comment by RWOverdijk** [comment]
It's not about what I _should_ or _shouldn't_ expect. It's about sharing thoughts and ideas and looking for the proper way forward, isn't it? We want this to work across platforms and so I'm sharing my thoughts 😄

Anyway, to be honest, I don't appreciate you telling me to be patient. At no point did I indicate I'm impatient. I've moved on from this library for about 3 weeks now. I'm not one to be patient. I'd rather get my hands dirty. That's why we love open source, right? 😄

This issue has the label "good-first-issue", are there any pointers for anyone to pick this up?

**Comment by edi** [comment]
What you do or don't appreciate is honestly not my problem. You are saying `Expo Payments` is in an usesless state, end quote. If you came here crying about the state of a feature which is meant for an API that's currently in **beta**, saying it's useless, then go on, create a PR and show us all 👍 

It's annoying to see people coming in, complaining about the state of a product they got access to for free.

Workarounds have been proposed by @nandorojo as you can clearly see in the previous posts. They work just fine, hence your comment is pointless.

Peace out.

**Comment by RWOverdijk** [comment]
> What you do or don't appreciate is honestly not my problem. You are saying Expo Payments is in an usesless state, end quote. If you came here crying about the state of a feature which is meant for an API that's currently in beta, saying it's useless, then go on, create a PR and show us all 👍

lol. Let's find common ground. I think I understand your (how I perceive it anyway) frustration with me, but I think it's all a big misunderstanding.

> It's annoying to see people coming in, complaining about the state of a product they got access to for free.

Agreed. I hate this as well. We don't know each other, so it makes sense that you'd think that's all I'm doing here. But that's really not the case, I don't like waiting. I like contributing and I do so regularly.

> Workarounds have been proposed by @nandorojo as you can clearly see in the previous posts. They work just fine, hence your comment is pointless.
>
>Peace out.

I get that, but the workaround doesn't work around the mobile issue, unfortunately. Therefor I think the module is currently in a useless state (I keep adding "I think, to clarify that I'm not sure if what I'm saying is correct). Like I said, this issue has the label "good-first-issue", are there any pointers for anyone to pick this up? Right now there's limited info on a desired solution making it hard to do anything other than thinking in workarounds.

> EDIT: You saying "I've moved on", and then complaining about the state of the feature less than a week ago, is just funny 🥇

I did move on from this library, but that was not a complaint. I think it's a misunderstanding. I state something I think to be true, and expect some shape of feedback. In my comment I shared that working with a webview and stripe elements allows you to use stripe on mobile **and** web quite easily. I was providing a workaround for both platforms, and felt like nandorojo thought my comment was directly linked to his. Which it wasn't. So I clarified (but, apparently not good enough).

When I said: _I use elements in a webview now, too. _, I meant that besides using elements for web, I also use elements in a webview as well. And it works.

Bottom line, not complaining. But I'm still interested in a solution because this is likely not the last project where I'll need to implement a solution for stripe.

Hope this clears things up and we can be nice to each other again. <3

**Comment by kopax** [comment]
Hello, just to get an update status on this. in may 2020, after (I hope so) the world pandemic, isn't there no new solution done while waiting home for this? I want to avoid ejecting and after a few hours of research and reading, I am thinking to use expo-stripe http client and make my own elements, or do a web view for target ios native and web. It's unclear now and I feel there's just old information out there. Thanks a lot and keep it cool!

**Comment by andresribeiro** [comment]
> Okay, that's nice. But who uses expo and doesn't also run on mobile? The whole point of this is to have support across platforms, isn't it?

@RWOverdijk Yes, a great advantage is having support on several platforms, but in some cases I just want to export a project as a PWA so I don't have to pay the fees for the App Store and Google Play Store, besides I have more control of the application, without worrying the policy of Apple and Google. I mean, waiting for updates and acceptance of the app is something I don't like, and in some cases 30% on all transactions is a price that can make the product unfeasible.

**Comment by louiechristie** [comment]
Any plans to integrate with frictionless, ad free payments ([as predicted by Nicolas Negroprente the 1990's](https://www.wired.com/1995/02/negroponte-27/)), using the new [Web Monetization API](https://webmonetization.org/specification.html,)? Using the new [Coil](https://coil.com/) and [Puma Browser](https://www.pumabrowser.com/)? 

Anyone want to sponsor me to do it?

**Comment by akilisosa** [comment]
So I have been looking for a monetization solution for my application which is cross browser, and build for mobile. I haven't had a chance to look at nandorojo thing yet. but is it possible to support browsers with a different solution like paypal? or something else. does it have to be stripe?  

I don't favor one more than the other I was just wondering. 
Also do you think it would be possible to edit or remove those comments that are like pure drama...

**Comment by cruzach** [comment]
Since the `expo-payments-stripe` module was deprecated with the release of SDK 42, I'm going to close this issue. The web implementation should go in `@stripe/stripe-react-native`, rather than a different package, if the Stripe team decides to move forward with a web implementation.

Anyways, we collaborated with the team at Stripe to make the `@stripe/stripe-react-native` library work seamlessly in Expo projects, so we recommend you use that library instead. [Learn more in the API reference](https://docs.expo.io/versions/latest/sdk/stripe/), and check out this [excellent YouTube video that walks through integrating Stripe in your Expo app](https://www.youtube.com/watch?v=DZlAET7Tgx4&ab_channel=UnsureProgrammer). Thanks!
